syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/template-sources/connector/simple/pom.xml

```diff
@@ -68,10 +69,31 @@
 					</execution>
 				</executions>
 			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-enforcer-plugin</artifactId>
+				<version>3.0.0-M2</version>
+				<executions>
+					<execution>
+						<id>enforce-maven</id>
+						<goals>
+							<goal>enforce</goal>
+						</goals>
+						<configuration>
+							<rules>
+								<requireMavenVersion>
+									<version>[3.5,3.6)</version>
+									<message>Maven needs to be a 3.5.x version. If you are using built-in default Maven in Eclipse, you need to configure a specific Maven as the default embedded version is 3.6+. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details.</message>
```

Comment:
The initial message text seems a little repetitive to me. I would change the message slightly from: "Maven needs to be a 3.5.x version. If you are using built-in default Maven in Eclipse, you need to configure a specific Maven as the default embedded version is 3.6+. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details."

to: "The Maven version must be a 3.5.x version in Eclipse to work correctly. If using the default Maven version in Eclipse, which defaults to 3.6.x, you must configure a specific 3.5.x Maven instance manually. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details."

Also, I am wondering if this is this a temporary problem and we'll be getting rid of the JIRA link eventually?

Reply:
> Also, I am wondering if this is this a temporary problem and we'll be getting rid of the JIRA link eventually?

very low probability that we will get rid of it on short term given the amount of time allocated to investigate on it and that several people already had a look.
We will get rid of it when all dependencies will be fixed and upgraded. (Maybe Fuse 8?) Unfortunately again on this point, seems the root project is quite abandoned without a lot of maintenance resource allocated.

Reply:
> The Maven version must be a 3.5.x version in Eclipse to work correctly.

it is not specifically "in Eclipse" that it needs to be 3.5.x, it is for all Maven builds of Fuse Online extension.

> If using the default Maven version in Eclipse, which defaults to 3.6.x, you must configure a specific 3.5.x Maven instance manually. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details.

re-reading, the part I initially provided and yours makes me realize we could be more precise for this sentence on the "Maven in Eclipse".

With both previous comments, what do you think of the following proposal?

"The Maven version must be a 3.5.x version to work correctly. If using the embedded Maven in Eclipse, which defaults to 3.6.x, you must configure a specific 3.5.x Maven instance manually. See https://issues.jboss.org/browse/FUSETOOLS-3249 for more details."

Reply:
Yes! I agree with that last revision.